manim/scene/scene_file_writer.py

```diff
@@ -523,6 +536,29 @@ def combine_movie_files(self):
             os.remove(sound_file_path)
 
         self.print_file_ready_message(movie_file_path)
+        if file_writer_config["write_to_movie"] : 
+            for file_path in partial_movie_files:
+                # We have to modify the accessed time so if we have to clean the cache we remove the one used the longest.
+                modify_atime(file_path)
+
+    def clean_cache(self):
+        """Will clean the cache by removing the partial_move used by manim the longest ago."""
+        cached_partial_movies = [os.path.join(self.partial_movie_directory, file_name) for file_name in os.listdir(
+            self.partial_movie_directory) if file_name != "partial_movie_file_list.txt"]
+        if len(cached_partial_movies) > file_writer_config['max_file_cached']:
+            oldest_file_path = min(cached_partial_movies, key=os.path.getatime)
+            logger.info(
+                f"Partial movie directory is full. (> {file_writer_config['max_file_cached']} files). Manim removed the file used by manim the longest ago. ({os.path.basename(oldest_file_path)})."
+                + "You can change this behaviour by changing max_file_cached in config.")
+            os.remove(oldest_file_path)
+
+    def flush_cache_directory(self): 
+        """Delete all the partial movie files cached"""
```

Comment:
not required but this feels better
```suggestion
        """Delete all the cached partial movie files"""
```